modules/ppcp-settings/resources/js/data/common/hooks.js

```diff
@@ -43,6 +43,7 @@ const useHooks = () => {
 	const clientSecret = usePersistent( 'clientSecret' );
 	const isSandboxMode = usePersistent( 'useSandbox' );
 	const isManualConnectionMode = usePersistent( 'useManualConnection' );
+	const flags = usePersistent( 'flags' );
```

Comment:
We should create a new selector for this, since the "flags" (or "wooSettings") should not be part of Transient or Persistent response. They are read-only

```js
// Read-only values.
const wooSettings = useSelect( ( select ) => select( STORE_NAME ).wooSettings(), [] );
```

The selectors would look similar to this:

```js
// ...
export const transientData = ( state ) => {
	const { data, wooSettings, ...transientState } = getState( state ); // ← extract the wooSettings, to ensure they are not part of the "transientState" collection.
	return transientState || EMPTY_OBJ;
};

export const wooSettings = ( state ) => {
	return getState( state ).wooSettings || EMPTY_OBJ;
};
```